### PR TITLE
refactor: styled modal class selector

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- The reason why some of the modal changes when another smaller popup shows is the overriding of styles and having the same class selector.
- To make it easier to override the base styles, we now instead utilize inline tailwind classes, since CSS module usage has heavier selectors.
- Probably one thing to do as well is to refactor the existing usages of `global` in our CSS modules that tend to conflict with one another.
- Would love to hear some thoughts on this.

Preview:
image

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-437 #done

### CODE REVIEW

File: packages/shared/src/components/modals/StyledModal.tsx [hunk 1]
```diff
@@ -1,40 +1,31 @@
...
-        },
-      }}
-      className={classNames('focus:outline-none modal', contentClassName)}
+      portalClassName={className?.toString()}
+      overlayClassName={classNames(
+        'overlay flex fixed flex-col justify-center inset-0 max-h-[100vh] items-center px-5 bg-theme-overlay-quaternary z-[10]',
+        overlayClassName,
+      )}
+      className={classNames(
+        'focus:outline-none modal flex flex-col relative w-full max-w-[26.25rem] max-h-[100%] overflow-y-auto items-center bg-theme-bg-tertiary shadow-2 border border-theme-divider-secondary rounded-16',
```
Comment (sshanzel) [inline packages/shared/src/components/modals/StyledModal.tsx:25]: Fully converted the content of the styled modal css module

**Review by rebelchris — COMMENTED** [review]
Nice let's try the onboarding modal on iPhone first as this was where the fix was for initially.
I'll get back to you after testing it.

**Comment by sshanzel** [comment]
> Nice let's try the onboarding modal on iPhone first as this was where the fix was for initially.
I'll get back to you after testing it.

Nice, thank you for mentioning as I only tried it on android. I will let you know soon as I test it in iOS.

**Review by rebelchris — APPROVED** [review]
@sshanzel Just tested the flows on desktop and iOS chrome + safari, they seem solid from this side! 🥳

**Pull request merged**